core/arch/riscv/kernel/abort.c

```diff
@@ -0,0 +1,394 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022-2023 NXP
+ */
+
+#include <kernel/abort.h>
+#include <kernel/linker.h>
+#include <kernel/misc.h>
+#include <kernel/panic.h>
+#include <kernel/tee_ta_manager.h>
+#include <kernel/thread_private.h>
+#include <kernel/user_mode_ctx.h>
+#include <mm/core_mmu.h>
+#include <mm/mobj.h>
+#include <riscv.h>
+#include <tee/tee_svc.h>
+#include <trace.h>
+#include <unw/unwind.h>
+
+enum fault_type {
+	FAULT_TYPE_USER_MODE_PANIC,
+	FAULT_TYPE_USER_MODE_VFP,
+	FAULT_TYPE_PAGE_FAULT,
+	FAULT_TYPE_IGNORE,
+};
+
+#ifdef CFG_UNWIND
+
+/* Kernel mode unwind */
+static void __print_stack_unwind(struct abort_info *ai)
+{
+	struct unwind_state_rv state = {
+		.fp = ai->regs->s0,
+		.sp = ai->regs->sp,
+		.pc = ai->regs->epc,
+	};
+
+	print_stack_rv(&state, thread_stack_start(), thread_stack_size());
+}
+
+#else /* CFG_UNWIND */
+static void __print_stack_unwind(struct abort_info *ai __unused)
+{
+}
+#endif /* CFG_UNWIND */
+
+static __maybe_unused const char *abort_type_to_str(uint32_t abort_type)
+{
+	if (abort_type == ABORT_TYPE_DATA)
+		return "data";
+	if (abort_type == ABORT_TYPE_PREFETCH)
+		return "prefetch";
+	return "undef";
+}
+
+static __maybe_unused const char *
+fault_to_str(uint32_t abort_type, uint32_t fault_descr)
+{
+	/* fault_descr is only valid for data or prefetch abort */
+	if (abort_type != ABORT_TYPE_DATA && abort_type != ABORT_TYPE_PREFETCH)
+		return "";
+
+	switch (core_mmu_get_fault_type(fault_descr)) {
+	case CORE_MMU_FAULT_ALIGNMENT:
+		return " (alignment fault)";
+	case CORE_MMU_FAULT_TRANSLATION:
+		return " (translation fault)";
+	case CORE_MMU_FAULT_READ_PERMISSION:
+		return " (read permission fault)";
+	case CORE_MMU_FAULT_WRITE_PERMISSION:
+		return " (write permission fault)";
+	case CORE_MMU_FAULT_TAG_CHECK:
+		return " (tag check fault)";
+	default:
+		return "";
+	}
+}
+
+static __maybe_unused void
+__print_abort_info(struct abort_info *ai __maybe_unused,
+		   const char *ctx __maybe_unused)
+{
+	__maybe_unused size_t core_pos = 0;
+
+	if (abort_is_user_exception(ai))
+		core_pos = thread_get_tsd()->abort_core;
+	else
+		core_pos = get_core_pos();
+
+	EMSG_RAW("");
+	EMSG_RAW("%s %s-abort at address 0x%" PRIxVA "%s",
+		 ctx, abort_type_to_str(ai->abort_type), ai->va,
+		 fault_to_str(ai->abort_type, ai->fault_descr));
+	EMSG_RAW("cpu\t#%zu", core_pos);
+	EMSG_RAW("cause\t%016" PRIXPTR " epc\t%016" PRIXPTR,
```

Comment:
We're using a lower case "x" in arch/arm instead causing the hexadecimal values to use lower case letters. It would be nice to be consistent in OP-TEE even if this is a different architecture.